jerry-core/vm/vm.c

```diff
@@ -3535,6 +3546,13 @@ vm_run (const ecma_compiled_code_t *bytecode_header_p, /**< byte-code data heade
         const ecma_value_t *arg_list_p, /**< arguments list */
         ecma_length_t arg_list_len) /**< length of arguments list */
 {
+#ifdef VM_RECURSION_LIMIT
+  if (JERRY_UNLIKELY (JERRY_CONTEXT (vm_recursion_counter)-- == 0))
```

Comment:
You should throw an error forever if you reach 0.

Reply:
Good catch! Fixed.